src/client.ts

```diff
@@ -46,7 +45,12 @@ export class APIClient {
           delay: 5000,
           maxAttempts: 10,
           handleError: (err, context) => {
-            if (err.statusCode !== 429) context.abort();
+            if (
+              err.statusCode !== 429 ||
+              ([500, 502, 503].includes(err.statusCode) &&
+                context.attemptNum > 1)
+            )
+              context.abort();
```

Comment:
Code 500, 502, 503: Retry once
Code 429: Exponential Backoff ([No retry info](https://www.hexnode.com/mobile-device-management/developers/setting-up-an-api/status-and-error-codes/#:~:text=429%20%E2%80%93%20Too%20Many%20Request))